v2/observer/jobs.go

```diff
@@ -136,7 +150,8 @@ func (o *observer) syncJobPod(obj interface{}) {
 	}
 
 	if status.Phase == core.JobPhaseSucceeded ||
-		status.Phase == core.JobPhaseFailed {
+		status.Phase == core.JobPhaseFailed ||
+		status.Phase == core.JobPhaseTimedOut {
```

Comment:
@krancour how do we feel about adding a check for `JobPhaseUnknown` here as well?  Then, the `cancelTimer()` call can be consolidated here as well.  Or, was it a design decision to _not_ delete job resources if/when the phase is unknown?

Reply:
Good question. _In general_ I'm not quite sure what to do about unkown. It's a state we have only because I know it's _possible_ for the underlying pod to enter a state that k8s considers unkown. This is mostly a mystery to me. I've never seen it happen.

Let me think on this for a bit.